### PR TITLE
Launchpad: Hide design_edited task from task list if completed for post-launch sites.

### LINKED ISSUE #77292: Launchpad: Create "Edit home page design" task
The task to edit the design is already created. But we will still need to add the task to the "Keep building" task list once we have it.

The task is defined here:
fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Swrgcnpx%2Qzh%2Qjcpbz%2Qcyhtva%2Sfgntvat%2Siraqbe%2Snhgbznggvp%2Swrgcnpx%2Qzh%2Qjcpbz%2Sfep%2Ssrngherf%2Synhapucnq%2Synhapucnq.cuc%3Se%3Q8s5012q9%23171-og

### PR DESCRIPTION
**Pull request by sixhours**
Fixes Automattic/wp-calypso#77292

## Proposed changes:

* Using the filter introduced in #31186, this filters task visibility for the Keep Building list to only show `design_edited`/`site_edited` if it isn't completed.
* This keeps the task visible on the Stepper Launchpad regardless of completion status, but hides it from Keep Building's task list if completed.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Check out this patch to your sandbox and sandbox the API
* Create a new site from`/start` selecting "Promote myself or business" as a goal -- this will put you into the Build flow with the `build` site intent.
* When you get to Launchpad in Stepper, confirm that the "Edit site design" task is incomplete.
* Launch your site without completing the "Edit site design" task.
* Go to the developer console and GET from the WP REST API `wpcom/v2/sites/siteSlug/launchpad`
* You should see the following list of tasks:

Screen Shot 2023-06-08 at 11 50 02 AM

* GET from the WP REST API `wpcom/v2/sites/siteSlug/launchpad?checklist_slug=keep-building`
* You should see the following tasks:

Screen Shot 2023-06-08 at 12 01 42 PM

* Complete the Edit site design task in Stepper (click on it, go to the site editor, then navigate back to Launchpad in Stepper).
* Go back to the developer console and GET from the WP REST API `wpcom/v2/sites/siteSlug2/launchpad`
* You should see the same task list, with the `design_edited` task marked as complete:

Screen Shot 2023-06-08 at 12 09 46 PM

* GET from the WP REST API `wpcom/v2/sites/siteSlug/launchpad?checklist_slug=keep-building`
* You should see the Keep Building task list no longer has the `design_edited` task:

Screen Shot 2023-06-08 at 12 19 41 PM

### CODE REVIEW

**Comment by sixhours** [comment]
Looking at #31186 I think the changes introduced there could solve the hacky logic issue. Holding off on this until that gets merged.

File: projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad.php
Comment (valterlorran) [inline projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad.php]: I think would be nice to have a function to get this option, since we are using it constantly.

File: projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad.php
Comment (valterlorran) [inline projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad.php]: I think we should rename this parameter to `$task_list`.
```suggestion
function wpcom_launchpad_keep_building_visible_tasks( $task_list ) {
```

File: projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad.php
Comment (valterlorran) [inline projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad.php]: I didn't test it, but it twisted my brain a little. I think that if one of the tasks is complete, they will both be excluded.
```suggestion
	return array_filter(
		$task_ids,
		function ( $task_id ) {
			// Only show design_edited/site_edited if it hasn't been marked as complete.
			if ( in_array( $task_id, [ 'design_edited', 'site_edited' ], true ) ) {
				return ! is_array( wpcom_is_checklist_task_complete($task_id) );
			}

			return true;
		}
	);
```

**Comment by nelsonec87** [comment]
I tested it but got the `design_edited` task as completed in the end
image

**Comment by sixhours** [comment]
> I tested it but got the design_edited task as completed in the end

For some reason (I don't honestly know why, it's been so long since I looked at this 😆 ) I had wrapped the task completion check in an `is_array` check, which would have always returned false since the task completion is a boolean. Removed that and tested it and it's working for me.

**Review by andres-blanco — APPROVED** [review]
Works as described. :shipit:

**Pull request merged**
